products/azure_rm_batchaccount/api.yaml

```
@@ -0,0 +1,170 @@
+azure_version: '2018-12-01'
+objects:
+  - !ruby/object:Api::Resource
+    name: BatchAccount
+    api_name: Batch
+    base_url: NotUsedInAzure
+
+    azure_sdk_definition: !ruby/object:Api::Azure::SDKDefinition
+      provider_name: batch
+      go_client_namespace: batch
+      go_client: batchAccountClient
+      python_client_namespace: azure.mgmt.batch
+      python_client: BatchManagementClient.batch_account
+      create: !ruby/object:Api::Azure::SDKOperationDefinition
+        async: true
+        go_func_name: Create
+        python_func_name: create
+        request:
+          'resourceGroupName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: resourceGroups
+            python_variable_name: resource_group_name
+            go_variable_name: resourceGroup
+          'accountName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: batchAccounts
+            python_variable_name: account_name
+            go_variable_name: accountName
+          '/': !ruby/object:Api::Azure::SDKTypeDefinition::ComplexObject
+            go_type_name: Properties
+            go_variable_name: parameters
+          /location: !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            go_field_name: location
+          /tags: !ruby/object:Api::Azure::SDKTypeDefinition::ComplexObject
+            go_field_name: tags
+          /properties/autoStorage: !ruby/object:Api::Azure::SDKTypeDefinition::ComplexObject
+            go_field_name: autoStorage
+          /properties/autoStorage/storageAccountId: !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            go_field_name: storageAccountId
+          /properties/poolAllocationMode: !ruby/object:Api::Azure::SDKTypeDefinition::ComplexObject
+            go_field_name: poolAllocationMode
+          /properties/keyVaultReference: !ruby/object:Api::Azure::SDKTypeDefinition::ComplexObject
+            go_field_name: keyVaultReference
+          /properties/keyVaultReference/id: !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            go_field_name: id
+          /properties/keyVaultReference/url: !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            go_field_name: url
+        response:
+      update: !ruby/object:Api::Azure::SDKOperationDefinition
+        go_func_name: Update
+        python_func_name: update
+        request:
+          'resourceGroupName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: resourceGroups
+            python_variable_name: resource_group_name
+            go_variable_name: resourceGroup
+          'accountName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: batchAccounts
+            python_variable_name: account_name
+            go_variable_name: accountName
+          '/': !ruby/object:Api::Azure::SDKTypeDefinition::ComplexObject
+            go_type_name: Properties

```

Here Python SDK handles it differently than Go SDK. So all the fields below are only applicable to Go SDK. Python SDK flattens all the fields out to the functions parameters. You may ask Python SDK team for the flattening logic they use in their SDK.

```suggestion
            applicable_to: [go]
            go_type_name: Properties
```